neo4j/internal/bolt/bolt4.go

```diff
@@ -924,3 +951,24 @@ func (b *bolt4) ForceReset() error {
 	b.Reset()
 	return b.err
 }
+
+// TODO: skip JSON construction if TRACE is not enabled?
+func serializeTrace(v interface{}) string {
+	builder := strings.Builder{}
+	encoder := json.NewEncoder(&builder)
+	encoder.SetEscapeHTML(false)
+	_ = encoder.Encode(v)
+	return strings.TrimSpace(builder.String())
+}
+
+// TODO: return map as-is if TRACE is not enabled?
+func skipCredentials(hello map[string]interface{}) map[string]interface{} {
+	result := make(map[string]interface{}, len(hello))
+	for k, v := range hello {
```

Comment:
Use the "map as a map" to retrieve the value and replace it.

Reply:
Do you mean something like:
```
hello["credentials"] = "<redacted>"
```
?

Reply:
💡yep, much better :)